File: app/src/main/java/com/taxi/friend/taxifriendclient/repositories/UserRepository.java
package com.taxi.friend.taxifriendclient.repositories;

import com.taxi.friend.taxifriendclient.models.Location;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.PUT;
import retrofit2.http.Path;

public interface UserRepository {

    @PUT("users/{userId}/location")
    Call<String> updateLocation(@Path("userId") String driverId, @Body Location location);

}

File: app/src/main/java/com/taxi/friend/taxifriendclient/repositories/UserRepository.kt
package com.taxi.friend.taxifriendclient.repositories

import com.taxi.friend.taxifriendclient.models.Location

import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.PUT
import retrofit2.http.Path

interface UserRepository {

    @PUT("users/{userId}/location")
    fun updateLocation(@Path("userId") driverId: String, @Body location: Location): Call<String>

}
